src/XRP/model/transaction-result.ts

```diff
@@ -0,0 +1,27 @@
+import TransactionStatus from '../transaction-status'
+
+/**
+ * Represents the outcome of submitting an XRPL transaction.
+ */
+export default class TransactionResult {
+  hash: string
```

Comment:
please mark as `public readonly`. 

Immutability constraints are generally helpful when we can apply them because it makes client code easier to reason about.